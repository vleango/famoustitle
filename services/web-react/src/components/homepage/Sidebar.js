import React, {Component} from 'react';
import { connect } from 'react-redux';
import { Form, Input } from 'reactstrap';
import { size, reverse, map } from 'lodash';
import moment from 'moment';
import Spinner from '../shared/Spinner';

import fontawesome from '@fortawesome/fontawesome';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import faSearch from '@fortawesome/fontawesome-free-solid/faSearch';

import './css/sidebar.css';

fontawesome.library.add(faSearch);

export class Sidebar extends Component {

    constructor(props) {
        super(props);
        this.state = {
            search: ""
        };
    }

    componentWillReceiveProps = (nextProps) => {
        if(nextProps.selected) {
            let search = "";
            if(nextProps.selected.match) {
                search = nextProps.selected.match
            }
            this.setState({
                search: search
            });
        }
    };

    onInputChange = (e) => {
        const field = e.target.name;
        const value = e.target.value;
        this.setState(() => ({ [field]: value }));
    };

    onSearchSubmit = (e) => {
        e && e.preventDefault();
        this.props.updateFilter("match", this.state.search);
    };

    render() {
        return (
            <div className="search-container">
                <aside>
                    <Form className="search-form" onSubmit={this.onSearchSubmit}>
                        <Input className="form-control pr-5" type="search" name="search" placeholder="Search..." value={this.state.search} onChange={this.onInputChange} />
                        <button className="search-button" type="submit"><FontAwesomeIcon icon="search"/></button>
                    </Form>
                </aside>
                <aside className="widget">
                    <div className="widget-title">Archives</div>
                    { size(this.props.archives) === 0 && <Spinner/> }
                    <ul>
                        {
                            this.props.archives && reverse(map(this.props.archives, (count, date) => {
                                const momentDate = moment.utc(date);
                                let className = "btn btn-link widget--date";
                                if(this.props.selected && this.props.selected.date === momentDate.format("YYYY-MM-DD") ? "* " : "") {
                                    className += " widget__selected";
                                }
                                return <li key={date}>
                                    <button className={className} onClick={() => this.props.updateFilter("date", momentDate.format("YYYY-MM-DD"))}>{momentDate.format("MMMM YYYY")} ({count})</button>
                                </li>
                            }))
                        }
                    </ul>
                </aside>
                <aside className="widget">
                    <div className="widget-title">Tags</div>
                    { size(this.props.archives) === 0 && <Spinner/> }
                    <div className="tagcloud">
                        {
                            this.props.tags && this.props.tags.map((tag) => {
                                return <button key={tag} className={this.props.selected && this.props.selected.tag === tag ? "selected" : ""} onClick={() => this.props.updateFilter("tag", tag)}>{tag}</button>
                            })
                        }
                    </div>
                </aside>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        archives: state.articles.index.archives,
        tags: state.articles.index.tags,
        selected: state.articles.index.selected
    };
};

export default connect(mapStateToProps)(Sidebar);
