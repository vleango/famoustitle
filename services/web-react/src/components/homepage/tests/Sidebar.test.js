import React from 'react';
import { shallow } from 'enzyme';

import Sidebar from '../Sidebar';

describe('Components', () => {
  describe('HomePage', () => {
    describe('Sidebar', () => {

      let wrapper = shallow(<Sidebar />);

      describe('Snapshot', () => {
        it('should correctly render Sidebar', () => {
          expect(wrapper).toMatchSnapshot();
        });
      });
    });
  });
});
