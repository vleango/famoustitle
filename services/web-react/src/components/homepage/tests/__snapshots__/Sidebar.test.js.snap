// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components HomePage Sidebar Snapshot should correctly render Sidebar 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sidebar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <aside>
          <Form
                    className="search-form"
                    onSubmit={[Function]}
                    tag="form"
          >
                    <Input
                              className="form-control pr-5"
                              name="search"
                              onChange={[Function]}
                              placeholder="Search..."
                              type="search"
                              value=""
                    />
                    <button
                              className="search-button"
                              type="submit"
                    >
                              <FontAwesomeIcon$1
                                        border={false}
                                        className=""
                                        fixedWidth={false}
                                        flip={null}
                                        icon="search"
                                        listItem={false}
                                        mask={null}
                                        name=""
                                        pull={null}
                                        pulse={false}
                                        rotation={null}
                                        size={null}
                                        spin={false}
                                        symbol={false}
                                        transform={null}
                              />
                    </button>
          </Form>
</aside>,
        <aside
          className="widget"
>
          <div
                    className="widget-title"
          >
                    Archives
          </div>
          <ul />
</aside>,
        <aside
          className="widget"
>
          <div
                    className="widget-title"
          >
                    Tags
          </div>
          <div
                    className="tagcloud"
          />
</aside>,
      ],
      "className": "mr-5",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Form
            className="search-form"
            onSubmit={[Function]}
            tag="form"
>
            <Input
                        className="form-control pr-5"
                        name="search"
                        onChange={[Function]}
                        placeholder="Search..."
                        type="search"
                        value=""
            />
            <button
                        className="search-button"
                        type="submit"
            >
                        <FontAwesomeIcon$1
                                    border={false}
                                    className=""
                                    fixedWidth={false}
                                    flip={null}
                                    icon="search"
                                    listItem={false}
                                    mask={null}
                                    name=""
                                    pull={null}
                                    pulse={false}
                                    rotation={null}
                                    size={null}
                                    spin={false}
                                    symbol={false}
                                    transform={null}
                        />
            </button>
</Form>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Input
                className="form-control pr-5"
                name="search"
                onChange={[Function]}
                placeholder="Search..."
                type="search"
                value=""
/>,
              <button
                className="search-button"
                type="submit"
>
                <FontAwesomeIcon$1
                                border={false}
                                className=""
                                fixedWidth={false}
                                flip={null}
                                icon="search"
                                listItem={false}
                                mask={null}
                                name=""
                                pull={null}
                                pulse={false}
                                rotation={null}
                                size={null}
                                spin={false}
                                symbol={false}
                                transform={null}
                />
</button>,
            ],
            "className": "search-form",
            "onSubmit": [Function],
            "tag": "form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "form-control pr-5",
                "name": "search",
                "onChange": [Function],
                "placeholder": "Search...",
                "type": "search",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <FontAwesomeIcon$1
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon="search"
                  listItem={false}
                  mask={null}
                  name=""
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  transform={null}
/>,
                "className": "search-button",
                "type": "submit",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "border": false,
                  "className": "",
                  "fixedWidth": false,
                  "flip": null,
                  "icon": "search",
                  "listItem": false,
                  "mask": null,
                  "name": "",
                  "pull": null,
                  "pulse": false,
                  "rotation": null,
                  "size": null,
                  "spin": false,
                  "symbol": false,
                  "transform": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "button",
            },
          ],
          "type": [Function],
        },
        "type": "aside",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="widget-title"
>
              Archives
</div>,
            <ul />,
          ],
          "className": "widget",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Archives",
              "className": "widget-title",
            },
            "ref": null,
            "rendered": "Archives",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "ul",
          },
        ],
        "type": "aside",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="widget-title"
>
              Tags
</div>,
            <div
              className="tagcloud"
/>,
          ],
          "className": "widget",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Tags",
              "className": "widget-title",
            },
            "ref": null,
            "rendered": "Tags",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "tagcloud",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "aside",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <aside>
            <Form
                        className="search-form"
                        onSubmit={[Function]}
                        tag="form"
            >
                        <Input
                                    className="form-control pr-5"
                                    name="search"
                                    onChange={[Function]}
                                    placeholder="Search..."
                                    type="search"
                                    value=""
                        />
                        <button
                                    className="search-button"
                                    type="submit"
                        >
                                    <FontAwesomeIcon$1
                                                border={false}
                                                className=""
                                                fixedWidth={false}
                                                flip={null}
                                                icon="search"
                                                listItem={false}
                                                mask={null}
                                                name=""
                                                pull={null}
                                                pulse={false}
                                                rotation={null}
                                                size={null}
                                                spin={false}
                                                symbol={false}
                                                transform={null}
                                    />
                        </button>
            </Form>
</aside>,
          <aside
            className="widget"
>
            <div
                        className="widget-title"
            >
                        Archives
            </div>
            <ul />
</aside>,
          <aside
            className="widget"
>
            <div
                        className="widget-title"
            >
                        Tags
            </div>
            <div
                        className="tagcloud"
            />
</aside>,
        ],
        "className": "mr-5",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Form
              className="search-form"
              onSubmit={[Function]}
              tag="form"
>
              <Input
                            className="form-control pr-5"
                            name="search"
                            onChange={[Function]}
                            placeholder="Search..."
                            type="search"
                            value=""
              />
              <button
                            className="search-button"
                            type="submit"
              >
                            <FontAwesomeIcon$1
                                          border={false}
                                          className=""
                                          fixedWidth={false}
                                          flip={null}
                                          icon="search"
                                          listItem={false}
                                          mask={null}
                                          name=""
                                          pull={null}
                                          pulse={false}
                                          rotation={null}
                                          size={null}
                                          spin={false}
                                          symbol={false}
                                          transform={null}
                            />
              </button>
</Form>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Input
                  className="form-control pr-5"
                  name="search"
                  onChange={[Function]}
                  placeholder="Search..."
                  type="search"
                  value=""
/>,
                <button
                  className="search-button"
                  type="submit"
>
                  <FontAwesomeIcon$1
                                    border={false}
                                    className=""
                                    fixedWidth={false}
                                    flip={null}
                                    icon="search"
                                    listItem={false}
                                    mask={null}
                                    name=""
                                    pull={null}
                                    pulse={false}
                                    rotation={null}
                                    size={null}
                                    spin={false}
                                    symbol={false}
                                    transform={null}
                  />
</button>,
              ],
              "className": "search-form",
              "onSubmit": [Function],
              "tag": "form",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "form-control pr-5",
                  "name": "search",
                  "onChange": [Function],
                  "placeholder": "Search...",
                  "type": "search",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <FontAwesomeIcon$1
                    border={false}
                    className=""
                    fixedWidth={false}
                    flip={null}
                    icon="search"
                    listItem={false}
                    mask={null}
                    name=""
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    transform={null}
/>,
                  "className": "search-button",
                  "type": "submit",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "border": false,
                    "className": "",
                    "fixedWidth": false,
                    "flip": null,
                    "icon": "search",
                    "listItem": false,
                    "mask": null,
                    "name": "",
                    "pull": null,
                    "pulse": false,
                    "rotation": null,
                    "size": null,
                    "spin": false,
                    "symbol": false,
                    "transform": null,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "button",
              },
            ],
            "type": [Function],
          },
          "type": "aside",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="widget-title"
>
                Archives
</div>,
              <ul />,
            ],
            "className": "widget",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Archives",
                "className": "widget-title",
              },
              "ref": null,
              "rendered": "Archives",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "ul",
            },
          ],
          "type": "aside",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="widget-title"
>
                Tags
</div>,
              <div
                className="tagcloud"
/>,
            ],
            "className": "widget",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Tags",
                "className": "widget-title",
              },
              "ref": null,
              "rendered": "Tags",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "tagcloud",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "aside",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
